.github/workflows/ci.yml

```diff
@@ -30,6 +30,34 @@ jobs:
           path: |                                                                                                                                                                                             
             ./build/${{ env.PACKAGE_NAME }}
   
+  update-doxygen:
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/main'  
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        persist-credentials: false
+        fetch-depth: 0
+    - name: Create local changes
+      run: |
+        sudo apt-get install doxygen
+        cd docs
+        doxygen
+        cp -r ./html/* ./
+        rm -rf ./html
+        git add .
+    - name: Commit files
+      run: |
+        git config --local user.email "41898282+github-actions[bot]@users.noreply.github.com"
+        git config --local user.name "github-actions[bot]"
+        git commit -m "Add changes" -a
```

Comment:
Can we make this commit message a little more descriptive? Something like "Auto-generate Doxygen documentation via Github Actions" would probably be sufficient